Artesian/Artesian.SDK/Service/Clients/Client.cs

```diff
@@ -147,21 +148,59 @@ public Client(IArtesianServiceConfig config, string Url, ArtesianPolicyConfig po
+                            var responseDeserialized = await res.Content.ReadAsAsync<object>(_formatters, ctk);
 
                             if (res.StatusCode == HttpStatusCode.BadRequest)
                             {
-                                var responseDeserialized = await res.Content.ReadAsAsync<object>(_formatters, ctk);
+                                if (res.Content.Headers.ContentType.MediaType == "application/problem+json")
+                                {
+                                    ArtesianSdkProblemDetail problemDetail = JsonConvert.DeserializeObject<ArtesianSdkProblemDetail>(responseDeserialized.ToString());
+
+                                    if (problemDetail.Detail != null)
+                                    {
+                                        responseText = problemDetail.Detail;
+                                    }
+                                    else if (problemDetail.Extensions["Errors"].Count() > 0)
+                                    {
+                                        responseText = "";
+                                        for (int i = 0; i < problemDetail.Extensions["Errors"].Count(); i++)
+                                        {
+                                            responseText += problemDetail.Extensions["Errors"][i]["Value"][0]["ErrorMessage"];
+                                        }
+                                    }
+                                    else if (problemDetail.Title != null)
+                                    {
+                                        responseText = problemDetail.Title;
+                                    }
+                                    throw new ArtesianSdkValidationException(exceptionMessage + responseText, problemDetail);
+                                }
+
                                 responseText = _tryDecodeText(responseDeserialized);
 
                                 throw new ArtesianSdkValidationException(exceptionMessage + responseText);
                             }
 
+                            if (res.Content.Headers.ContentType.MediaType == "application/problem+json")
+                            {
+                                ArtesianSdkProblemDetail problemDetail = JsonConvert.DeserializeObject<ArtesianSdkProblemDetail>(responseDeserialized.ToString());
+
+                                if (problemDetail.Detail != null)
+                                {
+                                    responseText = problemDetail.Detail;
+                                }
+                                else if (problemDetail.Title != null)
+                                {
+                                    responseText = problemDetail.Title;
+                                }
+                                throw new ArtesianSdkValidationException(exceptionMessage + responseText, problemDetail);
```

Comment:
The issue explicitly states that the exception should contains the ProblemDetails instance if present.